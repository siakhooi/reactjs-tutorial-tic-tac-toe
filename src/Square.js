import React from 'react'
import PropTypes from 'prop-types'

Square.propTypes = {
  onClick: PropTypes.func,
  value: PropTypes.string
}

function Square (props) {
  return (
    <button
      className="square"
      onClick={props.onClick}
    >{props.value}</button>
  )
}

export default Square
