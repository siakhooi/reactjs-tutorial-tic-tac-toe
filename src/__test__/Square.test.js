import React from "react";
import renderer from 'react-test-renderer';
import Square from "../Square";

it("Square", () => {
  const dummyFunction = jest.fn();
  const component =
    renderer.create(
      <Square
        value={"1"}
        onClick={dummyFunction} />,
    );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onClick()
  expect(dummyFunction).toBeCalled();

});
