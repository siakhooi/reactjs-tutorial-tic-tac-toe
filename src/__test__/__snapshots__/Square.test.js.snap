// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square 1`] = `
<button
  className="square"
  onClick={[MockFunction]}
>
  1
</button>
`;
